sorc/chgres_cube.fd/model_grid.F90

```diff
@@ -609,71 +605,261 @@ subroutine define_input_grid_mosaic(localpet, npets)
+ kgds = 0
+ call gdt_to_gds(gfld%igdtnum, gfld%igdtmpl, gfld%igdtlen, kgds, i_input, j_input, res)
 
  ip1_input = i_input + 1
  jp1_input = j_input + 1
 
- polekindflag(1:2) = ESMF_POLEKIND_MONOPOLE
+ allocate(rlat(i_input,j_input))
+ allocate(rlon(i_input,j_input))
+ allocate(rlat_diff(i_input,j_input))
+ allocate(rlon_diff(i_input,j_input))
+ allocate(xpts(i_input,j_input))
+ allocate(ypts(i_input,j_input))
+ allocate(rlat_corner(ip1_input,jp1_input))
+ allocate(rlon_corner(ip1_input,jp1_input))
+ allocate(xpts_corner(ip1_input,jp1_input))
+ allocate(ypts_corner(ip1_input,jp1_input))
+
+ do j = 1, j_input
+ do i = 1, i_input
+   xpts(i,j) = float(i)
+   ypts(i,j) = float(j)
+ enddo
+ enddo
 
- print*,"- CALL GridCreate1PeriDim FOR INPUT GRID."
- input_grid = ESMF_GridCreate1PeriDim(minIndex=(/1,1/), &
+ print*,"- COMPUTE GRID CELL CENTER COORDINATES."
+ call gdswzd(kgds,1,(i_input*j_input),-9999.,xpts,ypts,rlon,rlat,nret)
+
+ if (nret /= (i_input*j_input)) then
+   call error_handler("GDSWZD RETURNED WRONG NUMBER OF POINTS.", 2)
+ endif
+
+ deallocate(xpts, ypts)
+
+ if (localpet == 0) then
+   print*,'after gdswzd lat/lon 11', rlat(1,1),rlon(1,1)
+   print*,'after gdswzd lat/lon ni/11', rlat(i_input,1),rlon(i_input,1)
+   print*,'after gdswzd lat/lon 11/nj', rlat(1,j_input),rlon(1,j_input)
+   print*,'after gdswzd lat/lon ni/nj', rlat(i_input,j_input),rlon(i_input,j_input)
+   print*,'after gdswzd lat/lon mid  ', rlat(i_input/2,j_input/2),rlon(i_input/2,j_input/2)
+ endif
+
+ do j = 1, jp1_input
+ do i = 1, ip1_input
+   xpts_corner(i,j) = float(i) - 0.5
+   ypts_corner(i,j) = float(j) - 0.5
+ enddo
+ enddo
+
+ print*,"- COMPUTE GRID CELL CORNER COORDINATES."
+ call gdswzd(kgds,1,(ip1_input*jp1_input),-9999.,xpts_corner,ypts_corner,rlon_corner,rlat_corner,nret)
+
+ if (nret /= (ip1_input*jp1_input)) then
+   call error_handler("GDSWZD RETURNED WRONG NUMBER OF POINTS.", 2)
+ endif
+
+ deallocate(xpts_corner, ypts_corner)
+
+ if (localpet == 0) then
+   print*,'after gdswzd lat/lon corner 11', rlat_corner(1,1),rlon_corner(1,1)
+   print*,'after gdswzd lat/lon corner ni/11', rlat_corner(ip1_input,1),rlon_corner(ip1_input,1)
+   print*,'after gdswzd lat/lon corner 11/nj', rlat_corner(1,jp1_input),rlon_corner(1,jp1_input)
+   print*,'after gdswzd lat/lon corner ni/nj', rlat_corner(ip1_input,jp1_input),rlon_corner(ip1_input,jp1_input)
+   print*,'after gdswzd lat/lon corner mid  ', rlat_corner(i_input/2,j_input/2),rlon_corner(i_input/2,j_input/2)
+   print*,'after gdswzd max/min corner ',maxval(rlat_corner),minval(rlat_corner),maxval(rlon_corner),minval(rlon_corner)
+
+ if (gfld%igdtnum == 32769) then
+   where(rlon > 180.0) rlon = rlon - 360.0
+   where(rlon_corner > 180.0) rlon_corner = rlon_corner -  360.0
+ endif
+
+ do j = 1, j_input
+ do i = 1, i_input
+    rlat_diff(i,j) = rlat_corner(i,j) - rlat(i,j)
+    rlon_diff(i,j) = rlon_corner(i,j) - rlon(i,j)
+ enddo
+ enddo
+
+   ncid = 34
```

Comment:
And don't forget to remove the write of this diagnostic file.

Reply:
I think this file write can be removed now?